test/array_data_creation.hpp

```diff
@@ -0,0 +1,123 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or mplied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <numeric>
+
+#include "sparrow/array_data.hpp"
+#include "sparrow/data_traits.hpp"
+
+namespace sparrow::test
+{
+    /**
```

Comment:
Use normal comments in tests, as these functions will not be part of the documented api. (we dont want this to appear in the documentation)

Reply:
Can't we exclude the test folder from the parsing ?

Reply:
No idea but I suppose yes, but even if we do the doxygen stuffs is both misleading (as in suggests it should appear in the some doc or at least is "official api") and unnecessarilly noisy.